### PR TITLE
(Deprecated PR) Use a pool instead of a single connection, and reorganize the worker around that

### PR DESCRIPTION
**Pull request by ewjoachim**
Cf. #148

I've tried to redo and complete my previous approach without touching the "run multiple jobs at the same time" part.
I think the next step will be the separation between sync connector and async connector.

### Successful PR Checklist:
- [x] Tests
- [ ] *Documentation*: Need to write more of it
- [ ] Had a good time contributing? Let's say I'll be glad when this will be behind us.

### CODE REVIEW

File: procrastinate/app.py [hunk 1]
```diff
@@ -57,6 +60,14 @@ def __init__(
             A :py:func:`App.task` that has a custom "name" parameter, that is not
             imported and whose module path is not in this list will
             fail to run.
+        worker_timeout:
+            This parameter indicates the maximum duration (in seconds) procrastinate
+            workers wait between each database job pull. Job activity will be pushed
+            from the db to the worker, but in case the push mechanism fails somehow,
+            workers will not stay idle longer than the number of seconds indicated by
+            this parameters.
```
Comment (elemoine) [inline procrastinate/app.py]: s/parameters/parameter

File: procrastinate/aiopg_connector.py [hunk 2]
```diff
@@ -65,52 +51,112 @@ def _wrap_json(self, arguments: Dict[str, Any]):
...
+            Passed to aiopg. Default is False instead of True to avoid messing with
+            the global state.
+        cursor_factory (psycopg2.extensions.cursor):
+            Passed to aiopg. Default is :py:class:`psycopg2.extras.RealDictCursor`
+            instead of standard cursor. There is no identified use case for changing
+            this.
+        maxsize (int):
+            Passed to aiopg. Cannot be lower than 2, otherwise worker won't be
+            functionning normally (one connection for listen/notify, one for executing
+            tasks)
```
Comment (elemoine) [inline procrastinate/aiopg_connector.py]: Ending dot for consistency.

**Comment by ewjoachim** [comment]
@elemoine Did you finish your review ? Or did you stop midway through ? (I'm suprised not to see any more important fix)

**Comment by elemoine** [comment]
> @elemoine Did you finish your review ? Or did you stop midway through ? (I'm suprised not to see any more important fix)

I was actually finished! But with Travis being red I also thought this PR was still WIP. Please tell me if you need another review. I'd be happy look at your changes again.

File: tests/integration/test_healthchecks.py [hunk 3]
```diff
@@ -26,15 +26,3 @@ def checker(pg_connector):
...
-@pytest.mark.parametrize(
-    "method",
-    ["check_connection_async", "get_schema_version_async", "get_status_count_async"],
-)
-async def test_db_down(method):
-    bad_job_store = aiopg_connector.PostgresConnector(dsn="", dbname="a_bad_db_name")
-    checker = HealthCheckRunner(bad_job_store)
-
-    with pytest.raises(psycopg2.Error):
-        await getattr(checker, method)()
```
Comment (elemoine) [inline tests/integration/test_healthchecks.py]: Was this a temporary removal?

File: procrastinate/worker.py [hunk 4]
```diff
@@ -15,70 +15,109 @@ def __init__(
...
+                continue
+            await self.process_job(job=job)
+
+    async def process_job(self, job: jobs.Job) -> None:
+        job_context = job.get_context()
+        self.current_job_context = job_context
+        log_context: types.JSONDict = {
+            "worker_name": self.worker_name,
+            "job": job_context,
+        }
```
Comment (elemoine) [inline procrastinate/worker.py]: Previously the `log_context` had no `worker_name` key when `name` wasn't set or was `None` in the worker constructor. Now there always is a `worker_name` key, whose value might be `None`.

**Pull request closed without merging**